From 29b00295f06d9fba8959564c26b06b2530352185 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Thu, 6 Jun 2013 13:18:33 +0900
Subject: [PATCH 566/611] LTSI: usb: fixup generation conflict

LTSI 3.4.46 doesn't have devm_ioremap_resource(),
and usb_xxx_phy().
This patch fixup it as old version

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 drivers/usb/phy/phy-rcar-usb.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/phy/phy-rcar-usb.c b/drivers/usb/phy/phy-rcar-usb.c
index d7d96c9..1582d14 100644
--- a/drivers/usb/phy/phy-rcar-usb.c
+++ b/drivers/usb/phy/phy-rcar-usb.c
@@ -195,13 +195,13 @@ static int __devinit rcar_usb_phy_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 
-	reg0 = devm_ioremap_resource(dev, res0);
+	reg0 = devm_ioremap_nocache(dev, res0->start, resource_size(res0));
 	if (IS_ERR(reg0))
 		return PTR_ERR(reg0);
 
 	res1 = platform_get_resource(pdev, IORESOURCE_MEM, 1);
 	if (res1) {
-		reg1 = devm_ioremap_resource(dev, res1);
+		reg1 = devm_ioremap_nocache(dev, res1->start, resource_size(res1));
 		if (IS_ERR(reg1))
 			return PTR_ERR(reg1);
 	}
@@ -221,7 +221,7 @@ static int __devinit rcar_usb_phy_probe(struct platform_device *pdev)
 	priv->phy.shutdown	= rcar_usb_phy_shutdown;
 	spin_lock_init(&priv->lock);
 
-	ret = usb_add_phy(&priv->phy, USB_PHY_TYPE_USB2);
+	ret = usb_set_transceiver(&priv->phy);
 	if (ret < 0) {
 		dev_err(dev, "usb phy addition error\n");
 		return ret;
@@ -234,9 +234,7 @@ static int __devinit rcar_usb_phy_probe(struct platform_device *pdev)
 
 static int __devexit rcar_usb_phy_remove(struct platform_device *pdev)
 {
-	struct rcar_usb_phy_priv *priv = platform_get_drvdata(pdev);
-
-	usb_remove_phy(&priv->phy);
+	usb_set_transceiver(NULL);
 
 	return 0;
 }
-- 
1.8.2

